src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,501 @@
+
+        self.type = type
+        self.description: typing.Optional[str] = None
+
+
+class ApiGearInterface:
+    def __init__(self, name: str):
+        super().__init__()
+
+        self.name: str = name
+        self.properties: typing.Dict[str, ApiGearProperty] = {}
+
+        # TODO: implement handling for operations
+        # self.operations
+
+        # TODO: implement handling for signals
+        # self.signals
+
+
+class ApiGearModule:
+    def __init__(self):
+        super().__init__()
+
+        self.interfaces: typing.List[ApiGearInterface] = []
+        self.enumerations: typing.List[ApiGearEnumeration] = []
+        self.structures: typing.List[ApiGearStructure] = []
+
+
+def get_allowed_enum_literal(name: str) -> str:
+    """
+    Check if this is is an allowed literal name, if not add prefix.
+
+    Background:
+
+    In VSS '123' is a perfectly fine string literal, usable as allowed value for a string.
+    The current exporter (this file) translated it previously to 123 which is not a valid DSS IDL literal.
+    Adding an underscore as prefix makes the generated IDL ok, but then gives problems if generating for example
+    Python code by Eclipse Cyclone DDS idlc Python Backend.
+    By that reason we now add a regular character instead.
+    """
+    if name[0].isdigit():
+        return "d" + name
+    return name
+
+
+def get_apigear_datatype(t: str) -> typing.Optional[ApiGearType]:
+    is_array = False
+    apigear_type = None
+    if t.endswith("[]"):
+        is_array = True
+        t = t[0:-2]
+
+    if t in ["int8", "int16", "int32", "uint8", "uint16", "uint32"]:
+        apigear_type = ApiGearType(ApiGearBasicType.INT32.value)
+    elif t in ["int64", "uint64"]:
+        apigear_type = ApiGearType(ApiGearBasicType.INT64.value)
+    elif t == "float":
+        apigear_type = ApiGearType(ApiGearBasicType.FLOAT32.value)
+    elif t == "double":
+        apigear_type = ApiGearType(ApiGearBasicType.FLOAT64.value)
+    elif t == "boolean":
+        apigear_type = ApiGearType(ApiGearBasicType.BOOL.value)
+    elif t == "string":
+        apigear_type = ApiGearType(ApiGearBasicType.STRING.value)
+    else:
+        apigear_type = ApiGearType(t.replace(".", "_"))
+
+    if is_array and apigear_type is not None:
+        apigear_type.is_array = True
+
+    return apigear_type
+
+
+def node_name(node: VSSNode) -> str:
+    return node.qualified_name().replace(".", "_")
+
+
+def export_node(node, module: ApiGearModule, interface=None):
+    log.debug(f"Node {node.qualified_name()}")
+    if node.type == VSSType.BRANCH:
+        new_interface_name = node_name(node)
+        new_interface = ApiGearInterface(new_interface_name)
+        module.interfaces.append(new_interface)
+
+        for child in node.children:
+            export_node(child, module, new_interface)
+    else:
+        if node.allowed != "":
+            # create a new enum
+            if (node.datatype.value in ["string", "string[]"]):
```

Comment:
probably makes sense to use the available datatype enum

Reply:
Changed the comparison to datatype enum.